platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/BasePlatformUnitTests.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.test.fixtures.platform.TestPlatformContextBuilder;
+import com.swirlds.config.extensions.test.fixtures.TestConfigBuilder;
+import com.swirlds.platform.config.PathsConfig_;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.nio.file.Path;
+import java.util.function.Function;
+import org.junit.jupiter.api.io.TempDir;
+
+/**
+ * Platform level unit test base class for common setup and teardown.
+ */
+public abstract class BasePlatformUnitTests {
+
+    public static final String TEST_MARKER_FILE_DIRECTORY = "marker_files";
+
+    /**
+     * Temporary directory provided by JUnit
+     */
+    @TempDir
+    protected Path tempDir;
+
+    /**
+     * Creates a default platform context for the tests
+     */
+    protected PlatformContext createDefaultPlatformContext() {
+        return createPlatformContext(null, null);
+    }
+
+    /**
+     * Creates a platform context for the tests with the given builder modifications. Modifications are applied in the
+     * following order:
+     * <ol>
+     *     <li>The config modifications are applied</li>
+     *     <li>The temp directory is added to the config for marker files</li>
+     *     <li>The platform context builder modifications are applied</li>
+     * </ol>
+     * Any configuration set by the platform context builder modifying method overrides the configuration created by by the config modifier. Best practice is to set configuration through the config modifier and all other platform context variables through the platform contect modifier.
+     *
+     * @param platformContextModifier the function to modify the platform context builder
+     * @param configModifier          the function to modify the test config builder
+     * @return the platform context
+     */
+    protected PlatformContext createPlatformContext(
+            @Nullable final Function<TestPlatformContextBuilder, TestPlatformContextBuilder> platformContextModifier,
+            @Nullable final Function<TestConfigBuilder, TestConfigBuilder> configModifier) {
```

Comment:
As a general pattern, passing in configuration via lambdas is something I dislike.

Here's an idea. 

- Extend `TestPlatformContextBuilder` with a method `withRootDirectory(@NonNull Path rootDirectory)`
    - if set, then call into the configuration builder to update directories, as needed
    - eventually when we have a file system manager (hopefully not too far in the future), this will configure the file system manager
- Remove `TestPlatformContextBuilder.withConfiguration(Configuration)` and replace it with `withConfigurationBuilder(ConfigurationBuilder)`. Should be a straight forward refactor, but may touch many files.
- Remove the `createPlatformContext()` method from this class and replace it with `createPlatformContextBuilder()`. Within that method, call `withRootDirectory()` using the test directory.

As a part of this change, I'd strongly recommend reaching out to @hendrikebbers. I know he has put a lot of thought into how is the best way to set up unit test infrastructure. At the end of the day, hashgraph team owns hashgraph unit tests, but it's possible Hendrick may have a few tricks up his sleeves that will make our lives easier.